### PR TITLE
adding sa per component

### PR DESCRIPTION
**Pull request by Gregory-Pereira**
/cc @cooktheryan

### CODE REVIEW

File: controllers/securesign_controller.go [hunk 1]
```diff
@@ -53,6 +53,8 @@ const (
 
 	gitsignConsoleCliName        = "gitsign"
 	gitsignConsoleCliDescription = "gitsign is a CLI tool that allows you to digitally sign and verify git commits."
+
+	OperatorServiceAccountName = "rhtas-operator-controller-manager"
```
Comment (cooktheryan) [inline controllers/securesign_controller.go:57]: I think we can remove these

Comment (cooktheryan) [reply controllers/securesign_controller.go:57]: Any of the service account, role and binding in this specific file creation for the manager is done by the deployment

File: controllers/securesign_controller.go [hunk 2]
```diff
@@ -154,86 +151,6 @@ func (r *SecuresignReconciler) ensureSecureSign(
...
-				if err != nil {
-					return false, err
-				}
-			}
-			if err != nil {
-				return false, err
-			}
-		}
-
-		role := kubernetes.CreateRole(securesign.Namespace, constants.ServiceAccountName, labels(*securesign),
```
Comment (cooktheryan) [inline controllers/securesign_controller.go:185]: we will need to create two new common/utils/kubernetes files one for role and one for binding. 

The first role will need to be assigned to fulcio tuf and rekor
https://github.com/sigstore/scaffolding/blob/main/config/fulcio/certs/101-binding.yaml
https://github.com/sigstore/scaffolding/blob/main/config/rekor/createsecret/101-binding.yaml
 https://github.com/sigstore/scaffolding/blob/main/config/tuf/createsecret/101-binding.yaml and have the secret resources access

The next will need to be for ctlog
https://github.com/sigstore/scaffolding/blob/main/config/ctlog/certs/101-binding.yaml which looks to have both configmap and secrets

Comment (cooktheryan) [reply controllers/securesign_controller.go:185]: These actually dont need to be new files but we should follow the same procedures done by the monitoring enablement. For example, https://github.com/securesign/secure-sign-operator/blob/main/controllers/rekor/create.go#L157

File: config/manager/kustomization.yaml [hunk 3]
```diff
@@ -5,4 +5,4 @@ kind: Kustomization
 images:
 - name: controller
   newName: quay.io/grpereir/operator
-  newTag: 1.0.1
+  newTag: testing
```
Comment (bouskaJ) [inline config/manager/kustomization.yaml:8]: Can you avoid replacing the image and image tag? We update it almost in every PR. :-/

File: controllers/ctlog/utils/ctlog_deployment.go [hunk 4]
```diff
@@ -8,7 +8,10 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-func CreateDeployment(namespace string, deploymentName string, configName string, labels map[string]string) *appsv1.Deployment {
+func CreateDeployment(namespace string, deploymentName string, configName string, labels map[string]string, serviceAccountName string) *appsv1.Deployment {
+	if serviceAccountName == "" {
+		serviceAccountName = constants.ServiceAccountName
```
Comment (bouskaJ) [inline controllers/ctlog/utils/ctlog_deployment.go:13]: I think we can remove the shared account ( and also account creation from the `securesign_controller`)

File: controllers/rekor/create.go [hunk 5]
```diff
@@ -24,6 +26,7 @@ const (
 	ComponentName               = "rekor"
 	rekorMonitoringRoleName     = "prometheus-k8s-rekor"
 	rekorServiceMonitorName     = "rekor-metrics"
+	RekorServiceAccountName     = "rekor-sa"
```
Comment (bouskaJ) [inline controllers/rekor/create.go:29]: I don't think we need to export `RekorServiceAccountName` -> change to lowercase `rekorServiceAccountName`

File: controllers/trillian/create.go [hunk 6]
```diff
@@ -21,7 +22,10 @@ const (
 	logserverDeploymentName = "trillian-logserver"
 	logsignerDeploymentName = "trillian-logsigner"
 
-	ComponentName = "trillian"
+	dbServiceAccountName        = "trillian-db-sa"
+	logsignerServiceAccountName = "trillian-logsigner-sa"
```
Comment (bouskaJ) [inline controllers/trillian/create.go:26]: I think we can share one SA for all deployments.

File: controllers/rekor/create.go (hunk 5 above)
Comment (Gregory-Pereira) [reply controllers/rekor/create.go:29]: Ill swap it back but I was under the assumption that the capital letters implied that they were constants

Comment (bouskaJ) [reply controllers/rekor/create.go]:
```
In Go, a name is exported if it begins with a capital letter. For example, Pizza is an exported name, as is Pi, which is exported from the math package.

pizza and pi do not start with a capital letter, so they are not exported.
```
See https://go.dev/tour/basics/3

**Review by bouskaJ — COMMENTED** [review]
Few more comments but otherwise looks good :+1:

**Comment by bouskaJ** [comment]
resolved by https://github.com/securesign/secure-sign-operator/pull/95

**Pull request closed without merging**